core/Cargo.toml

```diff
@@ -16,31 +16,41 @@ async-trait = "0.1.57"
 clap = { version = "4.0.9", features = ["derive"], optional = true }
 hex = { version = "0.4.3", default-features = false }
 log = "0.4.17"
-parity-scale-codec = "3.2.2"
+parity-scale-codec = { version = "3.6.1", features = ["derive"] }
 serde = "1.0.136"
 serde_json = "1.0.85"
 zstd = { version = "0.11.2", default-features = false }
 
 # Substrate
-frame-remote-externalities = { version = "0.28.0" }
-frame-try-runtime = { version = "0.27.0", features = ["try-runtime"] }
+frame-remote-externalities = { git = "https://github.com/paritytech/substrate", branch = "master" }
+frame-try-runtime = { git = "https://github.com/paritytech/substrate", branch = "master", features=["try-runtime"] }
 
-sc-cli = { version = "0.29.0" }
-sc-executor = { version = "0.25.0" }
+sc-cli = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sc-executor = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
-sp-api = { version = "19.0.0" }
-sp-core = { version = "20.0.0" }
-sp-externalities = { version = "0.18.0" }
-sp-io = { version = "22.0.0" }
-sp-keystore = { version = "0.26.0" }
-sp-runtime = { version = "23.0.0" }
-sp-rpc = { version = "19.0.0" }
-sp-state-machine = { version = "0.27.0" }
-sp-weights = { version = "19.0.0" }
+sp-api = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-version = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-core = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-externalities = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-io = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-keystore = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-rpc = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-state-machine = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-weights = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
-substrate-rpc-client = "0.26.0"
+substrate-rpc-client = { git = "https://github.com/paritytech/substrate", branch = "master" }
+
+[dev-dependencies]
+sc-service = { git = "https://github.com/paritytech/substrate", branch = "master" }
+assert_cmd = "2.0.10"
+regex = "1.7.3"
+tempfile = "3.1.0"
+tokio = "1.27.0"
+node-primitives = { git = "https://github.com/paritytech/substrate", branch = "master" }
+substrate-cli-test-utils = { git = "https://github.com/paritytech/substrate", branch = "master", features = ["try-runtime"]}
+node-executor = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
 [features]
-cli = [
-    "clap",
-]
+cli = ["clap"]
```

Comment:
if I'm not mistaken, now, we are not using this feature that much... surely it makes the code a bit cleaner, but it also creates a problem, that all theses `commands` submodules cannot be used in no-std environment (because every `pub struct Command` needs `clap`)

I'm not sure whether it is actually a problem, given that all these functionalities are now doing some networking etc... but my initial idea was that _if_ we have a very granular logic here, there might be some no-std/wasm usecases, where part of this logic would be useful (the part that is done after block fetching / snapshot creating / etc)... unfortunately, I cannot give now an existing example

Reply:
I think a lot of stuff would break if we tried to make this no-std, e.g. I don't think you can even use the `format!` macro. 

I'd hesitate to make this no-std compatible now which would hinder development a bit if there's no clear use case at this time.

What do you think about this: if there is some functionality down-the-line that would be useful in no-std environments, we create a seperate no-std crate in this workspace to encapsulate it?

Reply:
sure, let's don't care about it until somebody raises issues